content/publishing-yaml/distribution.md

```diff
@@ -227,3 +225,17 @@ And then export the filepath on the gradlew task
         cd android && ./gradlew appDistributionUploadRelease
 
 {{</notebox>}}
+
+### Publishing web applications to Firebase Hosting
+
+Publishing web applications to Firebase Hosting With Codemagic publishing to Firebase Hosting is a straight-forward process as the Firebase CLI is already pre-installed on our virtual machines. To get started, you will need to obtain your Firebase token. In order to do that, run `firebase login:ci` in your local terminal.
+
+After running the command, your default browser should prompt for authorization to your Firebase project - when access is granted, the necessary token will appear in your terminal. 
+
+The next step is to add your token to Codemagic. To properly set everything up, please [encrypt](https://github.com/codemagic-ci-cd/codemagic-docs/blob/master/content/building/encrypting) the token using Codemagic UI and set it under your environment variables with the name `FIREBASE_TOKEN`. Now all that is needed to publish to Firebase Hosting is to add the publishing step to your .yaml file
+
+    - name: Publish to Firebase Hosting
+      script: |
+        firebase deploy --token "$FIREBASE_TOKEN"
+
+Make sure to add the publishing step after the build step. After the build is finished, you can now find your application published to Firebase Hosting, it is also possible to retrieve the direct URL from the log output in the UI. Please note that before trying to publish to Firebase Hosting, you will have to set it up for your project locally. You can find more information in the official [documentation](https://firebase.google.com/docs/hosting/quickstart) for Firebase.
```

Comment:
When the build is successful, you can find your application published to Firebase Hosting. Note that it is also possible..

Can you also add a log example about the URL part?

Add the note about setting up Firebase Hosting to the beginning of the section instead